timesketch/lib/testlib.py

```diff
@@ -65,33 +65,66 @@ def __init__(self):
         """Initialize the client."""
         self.indices = MockOpenSearchIndices()
 
-    def search(self, index, body, size):  # pylint: disable=unused-argument
-        """Mock a client search, used for aggregations."""
-        meta = {
-            "es_time": 23,
-            "es_total_count": 5621,
-            "timed_out": False,
-            "max_score": 0.0,
+    def search(self, index, body, size=0, search_type=None):  # pylint: disable=unused-argument
+        """Mock a client search.
+
+        Used for testing both aggregations and adding event attributes.
```

Comment:
Is there a specific reason you are adding the test cases for adding attributes in that method vs running a separate `_add_event_attribute` method in `BaseTest`?

Reply:
Not sure I understand exactly, the endpoint makes calls to the opensearch client similar to `EventTaggingResource` so I've added a relevant mock response to `MockOpenSearchClient`. I don't see how adding an `_add_event_attribute` method to `BaseTest` would achieve the same thing but I may be misunderstanding.

Reply:
ACK